unlock-app/src/services/web3Service.js

```diff
@@ -47,45 +48,54 @@ export default class Web3Service extends EventEmitter {
    * @param {object} account
    * @return
    */
-  connect({ provider }) {
-    if (provider === this.provider) {
+  async connect({ provider: providerName }) {
+    if (providerName === this.provider) {
       // If the provider did not really change, no need to reset it
       return
     }
 
     // Keep track of the provider
-    this.provider = provider
+    this.provider = providerName
     // And reset the connection
     this.ready = false
 
     // We fail: it appears that we are trying to connect but do not have a provider available...
-    if (!providers[provider]) {
+    const provider = this.providers[providerName]
+    if (!provider) {
       return this.emit('error', new Error('Provider does not exist'))
     }
 
-    this.web3 = new Web3(providers[provider])
+    try {
+      if (provider.enable) {
```

Comment:
It would actually be worth testing if other types of wallet (on mobile for example) do implement a similar `enable` function.

Maybe worth trying if Brave and Opera (which both have crypto wallets in them) work?

Reply:
Brave wallet doesn't even show up in https://localhost:3000/provider and neither does Opera. I also can't find any documentation - at all - about how to develop for these wallets, so I'm unclear on where to even begin here. We can talk tomorrow.

Reply:
confirmed: all wallets we need to worry about provide "enable" which is basically metamask and opera. Brave uses metamask as its wallet in a chromium extension, mist and parity are still too difficult to install to have large market share. When that changes we can look at them again.